lib/internal/http2/compat.js

```diff
@@ -666,12 +670,13 @@ class Http2ServerResponse extends Stream {
   }
 
   [kFinish]() {
-    const stream = this[kStream];
-    const state = this[kState];
+    const stream = this;
+    const res = stream[kResponse];
+    const state = res[kState];
     if (state.closed || stream.headRequest !== state.headRequest)
       return;
     state.closed = true;
-    this[kProxySocket] = null;
+    stream[kProxySocket] = null;
```

Comment:
Note, `kProxySocket` should be cleared on the stream not the response. Should this be `null` or `undefined`? We set `kResponse` and `kRequest` to `undefiend`?